sycl/doc/SYCL_compiler_and_runtime_design.md

```diff
@@ -42,6 +42,91 @@ back-end. Today middle-end transformations include just a couple of passes:
 - **Back-end** - produces native "device" code in ahead-of-time compilation
 mode.
 
+### SYCL support in clang frontend
+
+SYCL support in clang frontend can be split into the following components:
+
+- Device code outlining. Since SYCL is a single source programming model
+compiler should be able to separate device code from host code.
+
+- Lowering of lambda function objects and named function objects ("SYCL kernel
+functions"). To execute "SYCL kernel functions" on OpenCL devices some
+transformations are required.
+
+- Device code diagnostics.
+
+- Integration header generation.
+
+#### Device code outlining
+
+Here is a code example of a SYCL program that demonstrates compiler outlining
+work:
+
+```C++
+int foo(int x) { return ++x; }
+int bar(int x) { throw std::exception("CPU code only!"); }
+...
+using namespace cl::sycl;
+queue Q;
+buffer<int, 1> a(range<1>{1024});
```

Comment:
`buffer<int, 1> a{range<1>{1024}};` to be coherent with other `{}` use in the example.

Reply:
Done, thanks!